src/concrete/ml/sklearn/base.py

```diff
@@ -1281,17 +1287,33 @@ def __init_subclass__(cls):
                 _TREE_MODELS.add(cls)
                 _ALL_SKLEARN_MODELS.add(cls)
 
-    def __init__(self, n_bits: int):
+    def __init__(self, n_bits: Union[int, Dict[str, int]]):
         """Initialize the TreeBasedEstimatorMixin.
 
         Args:
-            n_bits (int): The number of bits used for quantization.
+            n_bits (int, Dict[str, int]): Number of bits to quantize the model. If an int is passed
+                for n_bits, the value will be used for quantizing inputs and leaves. If a dict is
+                passed, then it should contain "op_inputs" and "op_leaves" as keys with
+                corresponding number of quantization bits so that:
+                    - op_inputs (mandatory): number of bits to quantize the input values
+                    - op_leaves (optional): number of bits to quantize the leaves
+                Default to 6.
         """
-        self.n_bits: int = n_bits
+
+        # Check if 'n_bits' is a valid value.
+        _inspect_tree_n_bits(n_bits)
+
+        #: The number of bits to quantize the model.
```

Comment:
What are these `#:` ? Do they mean something special?

Reply:
Good question,  it's a comment style, for class attribute.
You can find them all over our code.

Reply:
it's for sphinx (since we are still supporting it), it enables documenting instances attributes that are not defined by the init  (the docstring only documents init parameters) : 
- https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoproperty
- https://stackoverflow.com/questions/8649105/how-to-show-instance-attributes-in-sphinx-doc

works also for class attributes, although in this case we can simply have a `Attributes:` section in the class' docstring (ex: `_is_a_public_cml_model` in `BaseEstimator`) 

although you are right, in this case it shouldn't really appear as the n_bits is defined in the init's docstring so the comment could be removed